Regression: LSP sends unparsable message, breaking text editor lsp integration.

### Tested versions

- Not Reproducible in 4.3
- Reproducible in 4.4 dev

### System information

Godot v4.4.dev (efa144396) - Fedora Linux 41 (Workstation Edition) on Wayland - X11 display driver, Multi-window, 2 monitors - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3090 (nvidia; 565.77) - AMD Ryzen 9 5950X 16-Core Processor (32 threads)

### Issue description

My lsp (helix-lsp) leaves the following error log when trying to parse a message from Godot 4.4 LSP

helix_lsp::transport [ERROR] Exiting gdscript after unexpected error: Parse(Error("data did not match any variant of untagged enum ServerMessage", line: 0, column: 0))


### Steps to reproduce

1. Open Godot
2. Open gdscript file in project in helix using netcat to connect to lsp
3. Language server immediately exits

Issue does not exist in 4.3 stable


### Minimal reproduction project (MRP)

Any gdscript will crash it.

--- a/modules/jsonrpc/jsonrpc.cpp
+++ b/modules/jsonrpc/jsonrpc.cpp
@@ -122,6 +122,11 @@ Variant JSONRPC::process_action(const Variant &p_action, bool p_process_arr_elem
 		Variant id;
 		if (dict.has("id")) {
 			id = dict["id"];
+
+			// Account for implementations that discern between int and float on the json serialization level, by using an int if there is a .0 fraction. See #100914
+			if (id.get_type() == Variant::FLOAT && id.operator float() == (float)(id.operator int())) {
+				id = id.operator int();
+			}
 		}
 
 		if (object == nullptr || !object->has_method(method)) {

